mls-rs-identity-x509/Cargo.toml

```diff
@@ -13,7 +13,6 @@ default = ["std"]
 std = ["mls-rs-core/std", "dep:thiserror"]
 
 [dependencies]
-async-trait = "0.1.74"
```

Comment:
I think code generated by `maybe-async` needs this. But maybe it can be moved to a `[target.'cfg(mls_build_async)'.dependencies]` section.

Reply:
Thanks, I'll take a look at the build failures and see if I can fix them somehow!

Reply:
Thank you for your contribution and taking the time to clean up these dependencies! Don't hesitate if you have questions.

Reply:
Yes, your idea seems to work!